source/acquaman/AMScanActionControllerBasicFileWriter.cpp

```diff
@@ -78,8 +78,17 @@ void AMScanActionControllerBasicFileWriter::writeToFile(int fileRank, const QStr
 		qDebug() << "Stream (1) writing differential " << startTime.msecsTo(endTime) << " for size " << textToWrite.size();
 		*/
 
+		QFileInfo dataFileInfo(filePath_+".dat");
+		int sizeBeforeWrite = dataFileInfo.size();
+
 		AMTextStream dataStream(dataFile_);
 		dataStream << textToWrite;
+		dataStream.flush();
```

Comment:
In QT documentation, it said:
```
void QTextStream::flush ()

Flushes any buffered data waiting to be written to the device.
If QTextStream operates on a string, this function does nothing.
```

Actually, QTextStream won't write to device, if `flush()` is not called. Therefore, in order to compare the size of the file, we have to force `flush`.